tests/simulation.ts

```diff
@@ -40,14 +40,16 @@ const DEFAULT_TEST_SETTINGS = {
   message_cooldown: 5,
 };
 // constants
-const EMPTY_CHATTERS: User[] = [];
+const EMPTY_CHATTERS: Chatter[] = [];
 // async function type
 const AsyncFunction = (async () => {
   /* used for type information */
 }).constructor;
 
 // mock variables
-let mockChatters: User[] = [];
+let mockChatters: Chatter[] = [];
+let mockSubscribers: Chatter[] = [];
+let mockModerators: Chatter[] = [];
```

Comment:
I think we should not mock `twitchApi.getModerators()`, `twitchApi.getSubscribers()`, and `twitchApi.getChatters()` with `Promise<Chatter[]>` when the actual implementation returns `Promise<User[]>`.
I think we should change all these to `User[]`, and also have the `simSetChatters`, `simSetSubscribers`, and `simSetModerators` have `User[]` as arguments.
When mocking those values I think we should use the same type as the implementation.

Reply:
There was *some* reason I changed this, but my approach to testing changed and made this change unnecessary. And you're right, the interfaces should be the same, or the tests aren't really testing accurately.